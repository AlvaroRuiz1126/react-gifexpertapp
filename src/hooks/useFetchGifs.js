import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (categories) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    //con el useEffect, el componente solo se renderiza una unica vez o cuando cambie categories
    //los useEffect no pueden ser async
    useEffect(() => {
        getGifs(categories).then(imgs => 
                setState({
                    data: imgs,
                    loading: false
                })
        )
    }, [categories]);
    //console.log(state);


    // setTimeout(() => {
    //     setState({
    //         data: [1,2,3,4,5,6,7],
    //         loading: false
    //     });
    // }, 1000);

    return state;
};