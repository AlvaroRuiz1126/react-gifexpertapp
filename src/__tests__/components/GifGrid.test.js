import React from 'react'
import GifGrid from '../../Components/GifGrid'; '../../Components/GifGrid';
import { shallow } from "enzyme";
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGrid', () => {
    //let wrapper = shallow(<GifGrid categories={'naruto'} />);

    /* beforeEach(() => {
        wrapper = shallow(<GifGrid categories={'naruto'} />);
    }); */

    test('Hace match con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid categories={'naruto'} />);
        expect(wrapper).toMatchSnapshot();
    }); 

    test('Debe de mostrar items cuando se cargan iamgenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost',
            title: 'Lo que sea'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid categories={'naruto'} />);
        //No se quiere evaluar el snapShot
        expect(wrapper).toMatchSnapshot();
        //Se quiere evaluar si existe algun parrafo p, esto por si la data viene el texto loading desaparece
        expect(wrapper.find('p').exists()).toBe(false);
        //Ademas, al evaluar cuantos componentes renderiza, hace que sea una prueba mucha mas precisa
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
