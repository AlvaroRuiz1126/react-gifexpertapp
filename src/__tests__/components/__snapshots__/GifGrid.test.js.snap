// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe de mostrar items cuando se cargan iamgenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Lo que sea"
      url="https://localhost"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid Hace match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    naruto
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
